cli/resource.go

```diff
@@ -0,0 +1,298 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/odpf/salt/term" //nolint
+	entropyv1beta1 "go.buf.build/odpf/gwv/odpf/proton/odpf/entropy/v1beta1"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/odpf/salt/printer"
+	"github.com/spf13/cobra"
+)
+
+const outputJSON = "json"
+
+func cmdResource(ctx context.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "resource",
+		Aliases: []string{"resources"},
+		Short:   "Manage resources",
+		Annotations: map[string]string{
+			"group:core": "true",
+		},
+		Example: heredoc.Doc(`
+			$ entropy resource create
+			$ entropy resource list
+			$ entropy resource view
+			$ entropy resource delete
+			$ entropy resource edit
+		`),
+	}
+
+	cmd.AddCommand(
+		createResourceCmd(ctx),
+		listAllResourcesCommand(ctx),
+		viewResourceCommand(ctx),
+		editResourceCommand(ctx),
+		deleteResourceCommand(ctx),
+	)
+
+	return cmd
+}
+
+func createResourceCmd(ctx context.Context) *cobra.Command {
+	var filePath string
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "create a resource",
+		Example: heredoc.Doc(`
+			$ entropy resource create
+		`),
+		Annotations: map[string]string{
+			"action:core": "true",
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			spinner := printer.Spin("")
+			defer spinner.Stop()
+			cs := term.NewColorScheme()
+
+			var reqBody entropyv1beta1.CreateResourceRequest
+			if err := parseFile(filePath, &reqBody); err != nil {
+				return err
+			}
+			err := reqBody.ValidateAll()
+			if err != nil {
+				return err
+			}
+
+			client, cancel, err := createClient(cmd)
+			if err != nil {
+				return err
+			}
+			defer cancel()
+
+			res, err := client.CreateResource(ctx, &reqBody)
```

Comment:
We are using context sent as parameter and also the one from `cobra.Command`. What would be the need to do so? Let's figure out what are the things handled by `cobra.Command` context and see if it's possible to unify them.

Reply:
💯 

Refer <https://pkg.go.dev/github.com/spf13/cobra#Command.Context>

It looks we can do `root.ExecuteContext(ctx)` instead of `root.Execute()`  and the have context accessed through `cmd.Context()` everywhere. We can get rid of the `ctx` parameter to the `cmdXXXX` functions.